meshnets/utils/callbacks.py

```diff
@@ -51,6 +51,32 @@ def finalize(self, status: str) -> None:
 class GPUUsage(Callback):
     """Track and log GPU usage."""
 
+    def __init__(self, log_freq=50):
+        """
+        Args:
+            log_freq: The frequency, in batches, at which to log GPU usage.
+        """
+        self.log_freq = log_freq
+        self.gpu_usage_history = {}
+
+    def on_batch_end(self, trainer, pl_module):
+        if (trainer.global_step +
+                1) % self.log_freq == 0 or trainer.is_last_batch:
+            try:
+                all_gpus = GPUtil.getGPUs()
+                for gpu in all_gpus:
+                    trainer.logger.log_metrics(
```

Comment:
How will this look? All gpus together or separate info about them?

Reply:
Each GPU is logged individually so there is a metric "GPU_NAME_memory_usage" for each of them.